subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1804,7 +1804,13 @@ static ssize_t recvfrom_dtls_client(struct tls_context *ctx, void *buf,
 	switch (ret) {
 	case MBEDTLS_ERR_SSL_PEER_CLOSE_NOTIFY:
 		/* Peer notified that it's closing the connection. */
-		return 0;
+		ret = tls_mbedtls_reset(ctx);
```

Comment:
I can't return 0 for datagram socket as for stream, because that would be ambigous (it's ok to get 0 on datagram recv). And I won't introduce a custom errno value to indicate that server closed the session, IMO it'd be more confusing that using something from the standard errno space (especially that it was recently unified accross standard libraries we support in Zephyr).

`ECONNABORTED` seemed like a good choice to me to indicate that peer closed the session, especially that I changed other uses of this value to avoid ambiguity (as explained in the commit message). Alternatively, an option would be to use `ENOTCONN` to indicate that socket is no longer connected.

Reply:
> And I won't introduce a custom errno value to indicate that server closed the session, IMO it'd be more confusing that using something from the standard errno space

Yes (it would have been better to go with a separate API in that case).

> Alternatively, an option would be to use ENOTCONN to indicate that socket is no longer connected.

IMHO, that would be a better choice.